lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -0,0 +1,635 @@
+            except:
+                pass
+
+        if credentials.get('client_id') is not None or credentials.get('ad_user') is not None:
+            return credentials
+
+        return None
+
+    def _get_env_credentials(self):
+        env_credentials = dict()
+        for attribute, env_variable in AZURE_CREDENTIAL_ENV_MAPPING.iteritems():
+            env_credentials[attribute] = os.environ.get(env_variable, None)
+
+        if env_credentials['profile'] is not None:
+            credentials = self._get_profile(env_credentials['profile'])
+            return credentials
+
+        if env_credentials['client_id'] is not None:
+            return env_credentials
+
+        return None
+
+    def _get_credentials(self, params):
+        # Get authentication credentials.
+        # Precedence: module parameters-> environment variables-> default profile in ~/.azure/credentials.
+        
+        self.log('Getting credentials')
+
+        arg_credentials = dict()
+        for attribute, env_variable in AZURE_CREDENTIAL_ENV_MAPPING.iteritems():
+            arg_credentials[attribute] = params.get(attribute, None)
+
+        # try module params
+        if arg_credentials['profile'] is not None:
+            self.log('Retrieving credentials with profile parameter.')
+            credentials = self._get_profile(arg_credentials['profile'])
+            return credentials
+        
+        if arg_credentials['client_id'] is not None:
+            self.log('Received credentials from parameters.')
+            return arg_credentials
+        
+        # try environment
+        env_credentials = self._get_env_credentials()
+        if env_credentials:
+            self.log('Received credentials from env.')
+            return env_credentials
+
+        # try default profile from ~./azure/credentials
+        default_credentials = self._get_profile()
+        if default_credentials:
+            self.log('Retrieved default profile credentials from ~/.azure/credentials.')
+            return default_credentials
+
+        return None
+
+    def serialize_obj(self, obj, class_name):
+        '''
+        Return a JSON representation of an Azure object.
+
+        :param obj: Azure object
+        :param class_name: Name of the object's class
+        :return: serialized result
+        '''
+        serializer = Serializer()
+        return serializer.body(obj, class_name)
+
+    def get_poller_result(self, poller):
+        '''
+        Consistent method of waiting on and retrieving results from Azure's long poller
+
+        :param poller Azure poller object
+        :return object resulting from the original request
+        '''
+        try:
+            delay = 20
+            while not poller.done():
+                self.log("Waiting for {0} sec".format(delay))
+                poller.wait(timeout=delay)
+            return poller.result()
+        except Exception, exc:
+            self.log(str(exc))
+            raise Exception(str(exc))

```

Comment:
It's almost always better to do this instead:
``` python
except Exception as exc:
    self.log(str(exc))
    raise
```

By using raise all by itself in the exception handler, the exaclt exception you caught will be continued up the stack.  raising the exception again like this:
```
raise exc
```
Will change the stack trace and doing it as you have it here will additionally change the exception type (which can be good in some circumstances but probably isn't what you want here since you'd just be changing it to the Exception base class.)